### PR TITLE
Feature/simple wallet: DPP Verification Simple Wallet Component

### PR DESCRIPTION
**Pull request by matbmoser**
# Why we create this PR?
 
For executing the PoC of this feature https://github.com/eclipse-tractusx/sig-release/issues/655, it was needed a simple wallet that is able to sign credentials and verify them too. It should manage the credentials.
 
# What we want to achieve with this PR?
 
Have a simple wallet that can:
- Sign Credentials
- Generate JSONLD Contexts for SAMM Json Schemas
- Verify Credentials
- Manage Private and Public Keys
- Enable DID web methods
- Manage authentication and authorization
- Store the credentials
- Be decentral
 
# What is new?
 
## Added
- Added the simple wallet component built in python
- Added functionality to issue certified data credentials or any verifiable credential with datamodel v2
- Added functionality to verify credentials with JsonWebSignature2020
- Added DID method to visualize the public key
- Added api translator from SAMM json schema to JSON-LD Context
- Added bpn authorization mechanism with api keys
- Added storage of credentials
- Added private and public key managers.

## PR Linked to:
 https://github.com/eclipse-tractusx/sig-release/issues/655

### CODE REVIEW

**Comment by matbmoser** [comment]
I dont know why it removed this parts License and Security, I will re-add it

**Comment by matbmoser** [comment]
All the code was done by me and worked by me before here: https://github.com/matbmoser/simple-wallet and now in Tractus-X. So no IP checks are needed

File: .github/workflows/publish-dpp-verification-wallet-docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+env:
+  IMAGE_NAME: 'digital-product-pass-verification-wallet'
+  REGISTRY: 'ghcr.io'
+  IMAGE_NAMESPACE: 'tractusx'
+  COMMIT_SHA: ${{ github.sha }}
+  REPO_URL: ${{ github.server_url }}/${{ github.repository }}
+
+jobs:
+  build-on-docker:
+    if: github.repository == 'eclipse-tractusx/digital-product-pass'
```
Comment (saudkhan116) [inline .github/workflows/publish-dpp-verification-wallet-docker-image.yml:54]: remove the checks

File: .github/workflows/publish-dpp-verification-wallet-docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      - name: Update Docker Hub description
+        uses: peter-evans/dockerhub-description@v4
+        with:
+        # readme-filepath defaults to toplevel README.md, Only necessary if you have a dedicated file with your 'Notice for docker images'   
+          readme-filepath: docs/notice.md
+          username: ${{ secrets.DOCKER_HUB_USER }}
+          password: ${{ secrets.DOCKER_HUB_TOKEN }}
+          repository: ${{ env.IMAGE_NAMESPACE }}/${{ env.IMAGE_NAME }}
+
+  build-on-ghcr:
```
Comment (saudkhan116) [inline .github/workflows/publish-dpp-verification-wallet-docker-image.yml:113]: remove this block of ghcr, it is no more used and should not be referenced anywhere

File: .github/workflows/publish-dpp-verification-wallet-docker-image.yml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    if: github.repository != 'eclipse-tractusx/digital-product-pass'
```
Comment (saudkhan116) [inline .github/workflows/publish-dpp-verification-wallet-docker-image.yml:114]: remove the check

File: charts/digital-product-pass/templates/ingress-backend.yaml [hunk 4]
```diff
@@ -1,26 +1,26 @@
 #################################################################################
 # Tractus-X - Digital Product Passport Application
-  #
-  # Copyright (c) 2022, 2024 BMW AG, Henkel AG & Co. KGaA
```
Comment (saudkhan116) [inline charts/digital-product-pass/templates/ingress-backend.yaml:4]: There should be only year of creation in the license headers. BMW and Henkel should be in two different lines.
The updated license header should look like this:

Copyright (c) 2022 BMW AG,
Copyright (c) 2022 Henkel AG & Co. KGaA
Copyright (c) 2023 CGI Deutschland B.V. & Co. KG

File: dpp-backend/scripts/logging.ini [hunk 5]
```diff
@@ -2,7 +2,7 @@
 ; Tractus-X - Digital Product Passport Application
 ;
 ; Copyright (c) 2022, 2024 BMW AG, Henkel AG & Co. KGaA
-Copyright (c) 2023, 2024 CGI Deutschland B.V. & Co. KG
```
Comment (saudkhan116) [inline dpp-backend/scripts/logging.ini:5]: same as above

File: charts/digital-product-pass/templates/ingress-backend.yaml [hunk 6]
```diff
@@ -1,26 +1,26 @@
...
+#
+# Copyright (c) 2022, 2024 BMW AG, Henkel AG & Co. KGaA
 # Copyright (c) 2023, 2024 CGI Deutschland B.V. & Co. KG
```
Comment (saudkhan116) [inline charts/digital-product-pass/templates/ingress-backend.yaml:5]: see above

File: dpp-verification/charts/simple-wallet/LICENSE [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2022, 2024 BMW AG, Henkel AG & Co. KGaA
+   Copyright 2023, 2024 CGI Deutschland B.V. & Co. KG
```
Comment (saudkhan116) [inline dpp-verification/charts/simple-wallet/LICENSE:191]: The licenses look different here and in copyright headers, please check.

File: dpp-verification/charts/simple-wallet/templates/configmap.yaml [hunk 8]
```diff
@@ -0,0 +1,80 @@
+#################################################################################
+# Tractus-X - Digital Product Passport Application
+#
+# Copyright (c) 2022, 2024 BMW AG, Henkel AG & Co. KGaA
```
Comment (saudkhan116) [inline dpp-verification/charts/simple-wallet/templates/configmap.yaml:4]: see above

**Review by saudkhan116 — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-dpp-verification-wallet-docker-image.yml [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+name: "Publish Digital Product Pass Verification Simple Wallet Docker Image"
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+  push:
+    branches: [ "main", "develop", "feature/simple-wallet" ]
```
[the remark this answers is not in the document]
Comment (matbmoser) [reply .github/workflows/publish-dpp-verification-wallet-docker-image.yml:34]: Good

File: .github/workflows/publish-dpp-verification-wallet-docker-image.yml [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+    inputs:
+      branch:
+        description: "Branch to use"
+        required: true
+        default: 'main'
+        type: string
+
+env:
+  IMAGE_NAME: 'digital-product-pass-verification-wallet'
+  REGISTRY: 'ghcr.io'
```
[the remark this answers is not in the document]
Comment (matbmoser) [reply .github/workflows/publish-dpp-verification-wallet-docker-image.yml:47]: ok

**Comment by matbmoser** [comment]
Ok solve the workflow things you told me 
Question the ingress things are not going to be solved when merged?
Maybe we need to merge main on this branch so it gets the latest changes.

**Comment by matbmoser** [comment]
Ok @saudkhan116 I removed the ingress change so in the latest version with the ingress fixed there will be what you have already done.

**Review by saudkhan116 — CHANGES_REQUESTED** [review]
The license headers are still not yet correct as agreed

**Comment by saudkhan116** [comment]
@matbmoser, as per our discussion, we are now aligned and agreed on the copyright license headers. We will still use date range (creation date and the last updated date) to a changed file. Secondly, we make a separate copyright line for each company/code owner where applicable.

Example:

# Copyright (c) 2022, 2024 BMW AG
# Copyright (c) 2022, 2024 Henkel AG & Co. KGaA
# Copyright (c) 2023, 2024 CGI Deutschland B.V. & Co. KG
# Copyright (c) 2023, 2024 Contributors to the Eclipse Foundation
 ...
 
DPP-verification:
# Copyright (c) 2023, 2024 BMW AG
# Copyright (c) 2023, 2024 CGI Deutschland B.V. & Co. KG
# Copyright (c) 2024 Contributors to the Eclipse Foundation
...

File: dpp-verification/charts/simple-wallet/LICENSE (hunk 7 above)
Comment (saudkhan116) [reply dpp-verification/charts/simple-wallet/LICENSE:191]: Ultimately, we need to change license headers in all files. This change will be followed there

**Review by saudkhan116 — APPROVED** [review]

**Comment by matbmoser** [comment]
There was a bug in the "expiration date" check. Now its fixed

**Comment by matbmoser** [comment]
I needed to fix the chart version

**Pull request merged**